test/contract-factory.test.ts

```diff
@@ -0,0 +1,57 @@
+import { expect } from "chai";
+import { starknet } from "hardhat";
+import { TIMEOUT } from "./constants";
+import { StarknetContractFactory } from "hardhat/types/runtime";
+import { getPredeployedOZAccount } from "./util";
+import { OpenZeppelinAccount } from "@shardlabs/starknet-hardhat-plugin/dist/src/account";
+import * as fs from "fs";
+import { StarknetContract } from "@shardlabs/starknet-hardhat-plugin/dist/src/types";
+
+describe("StarknetContractFactory", function () {
+    this.timeout(TIMEOUT);
+    let contractFactory: StarknetContractFactory;
+    let account: OpenZeppelinAccount;
+    // let contract: StarknetContract;
+
+    before(async function () {
+        account = await getPredeployedOZAccount();
+        contractFactory = await starknet.getContractFactory("contract");
+
+        // contract = await account.deploy(contractFactory, { initial_balance: 0 });
+    });
+    it("should have file at getAbiPath()", async () => {
+        const abiPath = contractFactory.getAbiPath();
+        expect(fs.existsSync(abiPath)).to.be.true;
+    });
+    it("should have abi", async () => {
+        const { abi } = contractFactory;
+        expect(typeof abi).to.be.eq("object");
+        expect(abi).to.not.be.null;
+        expect(Object.keys(abi).length > 0).to.be.true;
+    });
+    it("should have increase_balance in abi", async () => {
+        const { increase_balance } = contractFactory.abi;
+        expect(typeof increase_balance).to.be.eq("object");
+        expect(increase_balance.type).to.be.eq("function");
+    });
+    it("should have file at metadataPath", async () => {
+        const { metadataPath } = contractFactory;
+        expect(fs.existsSync(metadataPath)).to.be.true;
+    });
+    it("should declare class", async () => {
+        // account.declare calls and return contractFactory.declare
```

Comment:
What about this?

Reply:
This one is just to make it clear that account.dec uses contractfac.dec

Reply:
Sorry, I mistook this for commented code xD. Makes sense to keep it.